components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx

```diff
@@ -254,41 +253,52 @@ export const Nfts = (props: Props) => {
     ]
   }, [searchValue, userNonSpamNfts, hiddenNfts, allSpamNfts, searchNfts])
 
-  const tabOptions = React.useMemo(() => {
-    const tabOptions: NftTabOption[] = [
+  const dropDownOptions: NftDropdownOption[] = React.useMemo(() => {
+    return [
       {
-        id: 'nfts',
-        label: getLocale('braveNftsTab'),
-        labelSummary: userNonSpamNfts.length || undefined
+        id: 'collected',
+        label: getLocale('braveNftsTabCollected'),
+        labelSummary: sortedNfts.length
       },
       {
         id: 'hidden',
         label: getLocale('braveNftsTabHidden'),
-        labelSummary: hiddenNfts.length || undefined
-      },
-      {
-        id: 'spam',
-        label: getLocale('braveNftsTabSpam'),
-        labelSummary: !isLoadingSimpleHashList ? allSpamNfts.length : ''
+        labelSummary: sortedHiddenNfts.concat(sortedSpamNfts).length
       }
     ]
+  }, [sortedHiddenNfts, sortedSpamNfts, sortedNfts])
+
+  const nftsAccountLookup = React.useMemo(() => {
+    const lookup: { [assetId: string]: BraveWallet.AccountInfo | undefined } = {}
+    sortedHiddenNfts
+      .concat(sortedSpamNfts)
+      .concat(sortedNfts)
+      .forEach((nft) => {
+        const account = accounts.find(
+          (account) =>
+            nft.coin === account.accountId.coin &&
+            new Amount(
+              getBalance(account.accountId, nft, tokenBalancesRegistry)
+            ).gte('1')
+        )
 
-    return tabOptions
-  }, [nftList, hiddenNfts, allSpamNfts, isLoadingSimpleHashList])
-
+        if (account) {
+          lookup[getAssetIdKey(nft)] = account
+        }
+      })
+    return lookup
+  }, [accounts, sortedHiddenNfts, sortedSpamNfts, sortedNfts, tokenBalancesRegistry ])
```

Comment:
This value is only used by `NFTGridViewItem`. Maybe run this computation inside of that component instead?

Reply:
Running this computation inside of each NFTGridView item is not performant because we would be running the computation n times. In this implementation, we ran it once and reuse the values for each NFT item rendered.

Reply:
```
.forEach((nft) => {
```